src/transformers/integrations.py

```diff
@@ -305,70 +307,95 @@ def dynamic_modules_import_trainable(*args, **kwargs):
+                name="huggingface-tune",
+                type="offline",
+                parameters=trainer.hp_space(None),
+                metrics=[dict(name="objective", objective=direction, strategy="optimize")],
+                parallel_bandwidth=1,
+                budget=n_trials,
+            )
 
-        experiment = sigopt.create_experiment(
-            name="huggingface-tune",
-            type="offline",
-            parameters=trainer.hp_space(None),
-            metrics=[dict(name="objective", objective=direction, strategy="optimize")],
-            parallel_bandwidth=1,
-            budget=n_trials,
-        )
+            logger.info(f"created experiment: https://app.sigopt.com/experiment/{experiment.id}")
 
-        logger.info(f"created experiment: https://app.sigopt.com/experiment/{experiment.id}")
+            for run in experiment.loop():
+                with run:
+                    trainer.objective = None
+                    trainer._hp_search_setup(run.run)
+                    if trainer.args.world_size > 1:
+                        torch.distributed.broadcast_object_list(pickle.dumps(trainer.args), src=0)
+
+                    trainer.train(resume_from_checkpoint=None)
+                    # If there hasn't been any evaluation during the training loop.
+                    if getattr(trainer, "objective", None) is None:
+                        metrics = trainer.evaluate()
+                        trainer.objective = trainer.compute_objective(metrics)
+                    run.log_metric("objective", trainer.objective)
+
+            best = list(experiment.get_best_runs())[0]
+            best_run = BestRun(best.id, best.values["objective"].value, best.assignments)
+        else:
+            from sigopt import Connection
+
+            conn = Connection()
+            proxies = kwargs.pop("proxies", None)
+            if proxies is not None:
+                conn.set_proxies(proxies)
+
+            experiment = conn.experiments().create(
+                name="huggingface-tune",
+                parameters=trainer.hp_space(None),
+                metrics=[dict(name="objective", objective=direction, strategy="optimize")],
+                parallel_bandwidth=1,
+                observation_budget=n_trials,
+                project="huggingface",
+            )
+            logger.info(f"created experiment: https://app.sigopt.com/experiment/{experiment.id}")
 
-        for run in experiment.loop():
-            with run:
+            while experiment.progress.observation_count < experiment.observation_budget:
+                suggestion = conn.experiments(experiment.id).suggestions().create()
                 trainer.objective = None
-                trainer.train(resume_from_checkpoint=None, trial=run.run)
+                trainer._hp_search_setup(suggestion)
+                if trainer.args.world_size > 1:
+                    torch.distributed.broadcast_object_list(pickle.dumps(trainer.args), src=0)
```

Comment:
This only works for standard DDP (not on TPU/SageMaker etc). So there should be an early test with a clear error message if we are in a different distributed setup.